addOns/automation/src/main/java/org/zaproxy/addon/automation/tests/UrlPresenceTest.java

```diff
@@ -0,0 +1,452 @@
+ *
+ * Copyright 2022 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.automation.tests;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+import org.apache.commons.httpclient.URI;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.model.SiteMap;
+import org.parosproxy.paros.model.SiteNode;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.addon.automation.AutomationEnvironment;
+import org.zaproxy.addon.automation.AutomationJob;
+import org.zaproxy.addon.automation.AutomationProgress;
+import org.zaproxy.addon.automation.gui.UrlPresenceTestDialog;
+import org.zaproxy.addon.automation.jobs.JobUtils;
+
+public class UrlPresenceTest extends AbstractAutomationTest {
+
+    public static final String TEST_TYPE = "url";
+
+    public static String PARAM_NAME = "name";
+    public static String PARAM_ON_FAIL = "onFail";
+    public static String PARAM_URL = "url";
+    private static String PARAM_OPERATOR = "operator";
+    public static String PARAM_REQUEST_HEADER_REGEX = "requestHeaderRegex";
+    public static String PARAM_REQUEST_BODY_REGEX = "requestBodyRegex";
+    public static String PARAM_RESPONSE_HEADER_REGEX = "responseHeaderRegex";
+    public static String PARAM_RESPONSE_BODY_REGEX = "responseBodyRegex";
+
+    private Data data;
+
+    public enum Operator {
+        or("or"),
+        and("and");
+
+        private final String value;
+
+        Operator(String value) {
+            this.value = value;
+        }
+
+        public String getOperatorValue() {
+            if (value.equalsIgnoreCase("and")) {
+                return Constant.messages
+                        .getString("automation.dialog.urlpresencetest.operator.and")
+                        .toString();
+            } else if (value.equalsIgnoreCase("or")) {
+                return Constant.messages
+                        .getString("automation.dialog.urlpresencetest.operator.or")
+                        .toString();
+            }
+            return Constant.messages
+                    .getString("automation.dialog.urlpresencetest.operator.or")
+                    .toString();
+        }
+    }
+
+    public UrlPresenceTest(String name, String onFail, AutomationJob job) {
```

Comment:
Why is this constructor here and why does it not do the same things as the one below?